design/open/lakefs-on-kv-testing-plan.md

```diff
@@ -0,0 +1,133 @@
+# lakeFS on KV - Testing Plan
```

Comment:
While in development we will also need to modify our github pipeline to run all of our tests in 2 flavors (kv enabled/disabled)
We need to decide on the following things:

1. How do we support these flavors (In Esti / in .github actions?)
2. Do we need to run all of our jobs in both flavors (maybe we can do some mitigations)
3. Will the kv flavor be blocking for merge oprerations (even on issues that are not part of KV) this can potentially block other teams development process due to a bug in our development

We will of course need this for the KV1 milestone as well

Reply:
1. Good question. I think we can run it as a separate workflow (similar to [Unit test Spark metadata client](https://github.com/treeverse/lakeFS/actions/workflows/spark.yaml), for both flavors, or better, for the flavor that is not currently the default, and have the default flavor run with the general testing package.
That way, during the early development stages, we will have the test running on Table DB as part of the general testing flow and with KV enabled on the dedicated workflow, and at some point, we can switch both flags, and have the KV kicking in to the general workflow, applying to all tests that are affected by the DB section that is being enabled, while we keep the Table DB at the side workflow, at least until we ditch it for good.
That way it will be easy to delete the irrelevant tests as it will be a dedicated workflow
2. Implied from (1) - only relevant jobs will be executed in the dedicated workflow. These can be the specific tests for the feature we are currently working on, or any related tests we want to add
3. I believe it should be blocking. Idealy, a bug in our development of the KV support, will not get through, as it will be blocked too

WDYT?